### PR TITLE
Refactor discovery tests

### PR DESCRIPTION
**Pull request by MaidaB**
- DiscovererMode renamed to DiscoveryMode
- Create DiscoveryTestCollection and DiscoveryTestContext
- Tests moved to a new class

### CODE REVIEW

File: e2e-tests/IIoTPlatform-E2E-Tests/Discovery/DiscoveryTestContext.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                return;
+            }
+            
+            // OutputHelper cannot be used outside of test calls, we get rid of it before a helper method would use it
+            OutputHelper = null;
+
+            // Remove servers
+            var applicationIds = ServersInfo.Select(s => s.applicationId?.ToString());
+            RemoveAllApplications(applicationIds.OfType<string>().ToList());        
+        }
```
Comment (tamaskleizer) [inline e2e-tests/IIoTPlatform-E2E-Tests/Discovery/DiscoveryTestContext.cs:42]: please call base.Dispose(true) at the end of the Dispose method

File: e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs [hunk 2]
```diff
@@ -149,6 +149,43 @@ internal static partial class TestHelper {
...
+        /// <param name="body">Body for the request</param>
+        /// <param name="queryParameters">Additional query parameters</param>
+        public static IRestResponse CallRestApi(
+            IIoTPlatformTestContext context,
+            Method method,
+            string route,
+            object body = null,
+            Dictionary<string, string> queryParameters = null
+        ) {
+            var cancellationTokenSource = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
```
Comment (tamaskleizer) [inline e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs:167]: please make it possible for tests to pass their own CancellationToken, so they can stick to their own timeout

**Review by cristipogacean — DISMISSED** [review]

**Comment by karok2m** [comment]
@MaidaB, can you please have a look at `lgtm-com` bot's report and resolve 2 new alerts ?

**Review by tamaskleizer — APPROVED** [review]

**Review by cristipogacean — APPROVED** [review]
:shipit:

**Pull request merged**